src/types.rs

```diff
@@ -741,6 +741,66 @@ where
             interiors: interiors,
         }
     }
+    /// Wrap-around previous-vertex
+    fn previous_vertex(&self, current_vertex: &usize) -> usize
+    where
+        T: Float,
+    {
+        (current_vertex + (self.exterior.0.len() - 1) - 1) % (self.exterior.0.len() - 1)
+    }
+}
+
+// used to check the sign of a vec of floats
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+enum ListSign {
+    Empty,
+    Positive,
+    Negative,
+    Mixed,
+}
+
+impl<T> Polygon<T>
+where
+    T: Float + Signed,
+{
+    /// Determine whether a Polygon is convex
+    // For each consecutive pair of edges of the polygon (each triplet of points),
+    // compute the z-component of the cross product of the vectors defined by the
+    // edges pointing towards the points in increasing order.
+    // Take the cross product of these vectors
+    // The polygon is convex if the z-components of the cross products are either
+    // all positive or all negative. Otherwise, the polygon is non-convex.
+    // see: http://stackoverflow.com/a/1881201/416626
+    pub fn convex(&self) -> bool {
```

Comment:
most things that return `bool` usually start with `is_` or `has_`. what do you think about calling this `is_convex`?